### PR TITLE
localstore: remove metaKey

### PR DESCRIPTION
**Pull request by disksing**
@ngaut @coocood

### CODE REVIEW

File: store/localstore/snapshot.go [hunk 1]
```diff
@@ -47,46 +47,50 @@ func newSnapshot(store *dbStore, ver kv.Version) *dbSnapshot {
...
-		if key.Cmp(k) != 0 { // currently on [7]
-			if exact {
-				return nil, nil, errors.Trace(kv.ErrNotExist)
-			}
-			// search for NextKey
-			key = k
+		// quick test for exact mode
+		if exact && (key.Cmp(k) != 0 || isTombstone(v)) {
+			return nil, nil, errors.Trace(kv.ErrNotExist)
+		}
```
Comment (coocood) [inline store/localstore/snapshot.go]: I think it's better to handle `excact` case completely here, then we know the following code is for none `exact` mode.
put complex condition check in one `if` is error prone.
```
if exact {
    if key.Cmp(k) != 0 || isTombstone(v) {
           return  nil, nil, errors.Trace(kv.ErrNotExist)
    }
    return k, v, nil
}

Comment (disksing) [inline store/localstore/snapshot.go]: @coocood 
 Your solution is much more clear. Thanks!

Comment (coocood) [inline store/localstore/snapshot.go]: LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**